Apps/Admin/Client/Store/Communications/CommunicationsEffects.cs

```diff
@@ -61,17 +63,22 @@ public async Task HandleAddAction(CommunicationsActions.AddAction action, IDispa
     {
         this.Logger.LogInformation("Adding communication");
 
-        ApiResponse<RequestResult<Communication>> response = await this.Api.Add(action.Communication).ConfigureAwait(true);
-        if (response.IsSuccessStatusCode && response.Content != null && response.Content.ResultStatus == ResultType.Success)
+        try
         {
-            this.Logger.LogInformation("Communication added successfully!");
-            dispatcher.Dispatch(new CommunicationsActions.AddSuccessAction(response.Content));
-            return;
+            RequestResult<Communication> response = await this.Api.Add(action.Communication).ConfigureAwait(true);
+            if (response.ResultStatus == ResultType.Success)
+            {
+                this.Logger.LogInformation("Communication added successfully!");
+                dispatcher.Dispatch(new CommunicationsActions.AddSuccessAction(response));
+            }
+        }
+        catch (Exception e) when (e is ApiException or HttpRequestException)
+        {
+            this.Logger.LogError("Error adding communication...{Error}", e);
+            RequestError error = StoreUtility.FormatRequestError(e);
+            this.Logger.LogError("Error adding communication, reason: {ErrorMessage}", error.Message);
+            dispatcher.Dispatch(new CommunicationsActions.AddFailAction(error));
         }
-
```

Comment:
There is no handling for the case where the response.ResultStatus in not successful. I think this code that was deleted is still needed for that case.

Reply:
Changed.